File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string pref = strs[0];
        int pref_len = pref.length();
        for(int i = 1 ;i<strs.size();i++) {
            string s = strs[i];
            while (pref_len > s.length() || pref != s.substr(0, pref_len)){
                pref_len--;
                if(pref_len==0){
                    return "";
                }
            pref = pref.substr(0, pref_len);
            }
        }
     return pref;
    }

};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        pref = strs[0]

        for i in range(1, len(strs)):
            while strs[i].find(pref) != 0:
                pref = pref[0 : len(pref)-1]

                if pref == "":
                    return ""
        return pref

        